# Single-Cell Visualization of Genes in the Head of Wildtype *Xenpus tropicalis*

# Reference

Sydney Arlis. unpublished Fall 2022 COSMOS Presentation. Elucidating the Function of the Clefting Gene *ISM1*. Manak and Houston Labs.  

# Introduction
Orofacial clefting (OFC) is a common genetic abnormality occurring 1 in every 1,000 births. It results after one or more facial primordia fail to fuse during development. This can present as cleft lip (CL), cleft palate (CP) or a mixture of both (CL/CP). These abnormalities can often be accompanied by difficulties eating and talking and social impacts including negative self-image and decreased social skills. Another Ph.D. student in my/the Manak lab is looking into the copy number variants (CNV) that could be causing these abnormalities. CNVs are structural variants that result in the amplification or deletion of a gene. The Manak lab has previously collected human patient samples and then used array-based comparative genomic hybridization (aCGH) to identify CNVs in individuals with OFC. These experiments identified the novel clefting gene isthmin 1 (*ISM1*), which, when knocked out in *Xenopus*, causes craniofacial dysmorphologies and in some cases induces a cleft. Although it is known that *ISM1* is important for proper craniofacial development, it is still unclear how perturbation of *ISM1* expression results in orofacial clefts. We predict that there will be a difference in the neural crest cells at stage 25 and in *ISM1* knockdowns, because the neural crest is not fully developed until 25 and *ISM1* should alter the neural crest because of its known interactions with craniofacial development. 

<img width="496" alt="Sydney SpringPlot" src="https://user-images.githubusercontent.com/124283697/235980852-6105f3f9-8c53-42b8-bf6f-77607d613eef.png">

Figure 1. Sydney Arlis unpublished data Fall 2022. Each dot represents a single cell in the head of a wildtype frog. The farther apart the dots are from each other, the more differentiated they are. The cells can be categorized into cell types based on how they have been grouped within the spring plot, the different colors arbitrarily represent these groupings. 

Figure 1, from Sydney Arlis, is a visualization of single cells that were identified in a wildtype *Xenopus* in the head/face region. Each dot is representative of a cell, and the colors align different groups of closely related cells. The lines connecting between the different points show their relation to each other. The different areas are showing where the cell was expressed, and as you move farther away from the center plot, the cells become more differentiated. These cells are being expressed during stage 25; most publications only show cells until stage 22. I am interested in doing a single-cell visualization to show wildtype neural crest development between stages 22 and 25. Current publications only show development up to stage 22, however, the neural crest is still forming until stage 25, so we want to see what different cell populations are present at this stage. It is just common practice to stop around stage 22, Sydney is just specifically focused on the formation of the neural crest, so we want to extend out to stage 25. Eventually we want to do this in an *ISM1* knockdown to see how neural crest formation is being affected.

# Materials and Methods
My colleague made her figure by taking single-cell RNA sequencing in the form of Fastq files and running them through a Python pipeline to map the transcripts to actual genes. We still have the Fastq files of data, and the filtered gene barcodes, matrices, and filters. I have stored the barcodes and filters on my GitHub repository, but the filtered matrices were still too big to be stored on GitHub, so they are available on the Huston lab server where I have full access to all the necessary files. The pipeline used is from the Allon Klein’s Lab’s GitHub page, https://github.com/AllonKleinLab/SPRING_dev. This pipeline uses our filtered sequences and plots them on the spring plot based on the determinants of how similar the cells are to one another. The program was used to determine how to group the cells based on which genes they are/are not expressing compared to other cells, based on how many groups it is told to separate into. 

First, I went onto terminal/Argon and downloaded python 3.10.10 as well as a few of its libraries, including miniconda 23.3.1. I also downloaded R 4.1.0 to be able to create the spring plot. Then, I will set up the SPRING data directory, with its many subdirectories to store all the data that will be produced. Each subdirectory also needed to contain certain input and output files to keep raw and matrix data separate and how to categorize each cell. Next, I ran the SPRING viewer 2.4.4 using the FastX terminal. I opened the terminal and loaded python to be able to view the data. Finally, using Louvain clustering, the program was able to generate cellular phenotype labels that can be written to SPRING. I made it so that cells with less than three genes are omitted from the count to help eliminate redundant genes and to get rid of a lot of the unnecessary differentiating genes that would distract and crowd the plot. The program then normalized the cell readings. By normalizing the counts, the SPRING plot can more accurately represent the similarities and differences between cells based on their gene expression profiles and allows for more accurate identification of cell types. After running the program, a mean variance plot is created as a quality control plot in single-cell RNA sequencing analysis. It shows the relationship between the mean and variance of gene expression across cells, and can be used to identify technical noise, batch effects, and other sources of variability in the data. Typically, the plot will show a curved shape, with higher variance observed for genes with lower mean expression. The slope of the curve can be used to estimate the degree of technical noise in the data, with steeper slopes indicating higher levels of noise. This shows what has been normalized and eliminated from the data that is used to create the SPRING plot. Finally, I navigated to http://localhost:8000/springViewer_1_6_dev.html?Trop to view the created plot. 

Once in SPRING viewer, I ran Louvain clustering to color code the groupings of the genes based on similarity. Louvain clustering is a uses a mathematical algorithm used to assign cells to clusters, so that the cells within a cluster are more like each other than they are to cells in other clusters. The algorithm works by optimizing a measure of cluster quality, so that the resulting clusters are as distinct from each other as possible. I decided to have them grouped into 30 clusters, Sydney only used 20 clusters. I opted to increase the number of clusters to see if there are differences in where neural crest cells show up and how the cells will relate to each other if there are more clusters. Finally, I went through and found which genes are enriched in each cluster to determine what cell types were grouped together.  

# Results 
 
 ![trop_full(2)](https://user-images.githubusercontent.com/124283697/235981009-b0702a43-6a02-48a0-af40-4ddbb5f04df1.png)

Figure 2: Made using SPRING viewer, there are 30 different groupings created, this was a few too many and there are many repeated groupings just with enhanced expression of one or more different genes than other groupings.
 
# Discussion and Conclusion

I produced Figure 2. There is always going to be some randomness to the program, which is why it does not look exactly like Sydney’s. When the program is giving each cell and gene their specific barcodes, there is a slight degree of randomness as to which barcodes are going to align with each other more so than others and how closely those genes are going to relate to other genes. Each color represents one of 30 different groupings, some groupings share genes, but are more enriched for a specific gene than the other groups. Determination of the multiple groups was done using ChatGPT and Xenbase. Using the SPRING viewer, I was able to see which genes were enriched in each of the 30 groupings, and then took that list and asked ChatGPT in which cells in *Xenopus tropicalis* are these genes being expressed. ChatGPT would give a list of cell types corresponding to these genes, that I would then double check for accuracy using XenBase (the go to website for all things *Xenopus* related). 

There are specific marker genes I used to determine the separate cell groupings, for instance we use SOX10 as a marker gene for neural crest cells. The three neural crest groupings had high levels of SOX10 enrichment, which is why they are all labeled as neural crests. We can then look for *ISM1* in these groupings and see that is more widely expressed in the green/cyan groupings and not in the yellow grouping (data not shown) farther from the other two. Even though there is randomness in the way the figure looks, we can still see that the groupings are alike to Sydney’s and are seemingly still similarly related to each other. Most of the cells are related to the brain, the blood, and proliferation. It makes sense that these would be the most abundant cell types in the head of the wildtype Xenopus tropicalis at stage 25. 

As I mentioned above, there are three separate groupings of neural crest cells. Our current working theory for this is that one grouping (the more yellow grouping) is not enriched for the *ISM1* gene, and is still forming the neural crest, while the other two groupings have *ISM1* expression and are finished proliferating or are in the final stages of forming the neural crest. This will need to be studied in greater detail, but both figures have helped us come to these conclusions. Next will be to visualize an *ISM1* knockdown model, against the wildtype model, to see what happens to these three diverging groups, and what happens when *ISM1* is no longer being expressed.

# Reflection

While I wouldn’t say that I encountered too many issues, I will say it was difficult for me to keep updating my GitHub repository. I felt as though I had no use for it. My data was too large to be stored there and I was not using it to hold any sort of analysis or outputs as I already had those from Sydney, and they also would have been far too large to store on GitHub. I did however, enjoy running the actual code to produce the figure, and learning from my PI (Doug Houston) how SPRING works and the different ways I can manipulate the figure. Honestly, just playing around in SPRING viewer was the most enjoyable part. There is a way to compare the genes between multiple groupings and this was helpful in comparing when there were multiple groupings for the same type of cells. For instance, there are multiple groupings for immune and blood cells, and the difference between them is usually only a couple of genes that were more enriched for one area than the other. I think this was also a great tool to help me learn where certain cells are being expressed, as I had to fully immerse myself within genetics to determine where the neural crest cells were being expressed, etc. 
